### PR TITLE
feat: FloatButton support dynamic z-index

### PR DESCRIPTION
**Pull request by li-jia-nan**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE_CN.md?plain=1)

### 🤔 This is a ...

- [x] 🆕 New feature
- [ ] 🐞 Bug fix
- [ ] 📝 Site / documentation improvement
- [ ] 📽️ Demo improvement
- [ ] 💄 Component style improvement
- [ ] 🤖 TypeScript definition improvement
- [ ] 📦 Bundle size optimization
- [ ] ⚡️ Performance optimization
- [ ] ⭐️ Feature enhancement
- [ ] 🌐 Internationalization
- [ ] 🛠 Refactoring
- [ ] 🎨 Code style optimization
- [ ] ✅ Test Case
- [ ] 🔀 Branch merge
- [ ] ⏩ Workflow
- [ ] ❓ Other (about what?)

### 🔗 Related Issues

> - Describe the source of related requirements, such as links to relevant issue discussions.
> - For example: close #xxxx, fix #xxxx

### 💡 Background and Solution

> - The specific problem to be addressed.
> - List the final API implementation and usage if needed.
> - If there are UI/interaction changes, consider providing screenshots or GIFs.

### 📝 Change Log

> - Describe the impact of the changes on developers, not the solution approach.
> - Reference: https://ant.design/changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | 将 FloatButton 组件的 `z-index` 加入 `useZIndex` 管理，兼容弹层类组件，并且将默认的 `z-index: 99` 样式替换为 `z-index: zIndexPopupBase` |
| 🇨🇳 Chinese | 将 FloatButton 组件的 `z-index` 加入 `useZIndex` 管理，兼容弹层类组件，并且将默认的 `z-index: 99` 样式替换为 `z-index: zIndexPopupBase` |

### CODE REVIEW

File: components/float-button/FloatButton.tsx [hunk 1]
```diff
@@ -59,11 +60,13 @@ const InternalFloatButton = React.forwardRef<FloatButtonElement, FloatButtonProp
     },
   );
 
+  // ============================ zIndex ============================
+  const [zIndex] = useZIndex('FloatButton', style?.zIndex as number);
+
+  const mergedStyle: React.CSSProperties = { zIndex, ...style };
```
Comment (zombieJ) [inline components/float-button/FloatButton.tsx:66]: 顺序写反了，现在这两行代码等价于：

```tsx
const mergedStyle = style;
```

File: components/float-button/FloatButtonGroup.tsx [hunk 2]
```diff
@@ -47,6 +48,11 @@ const FloatButtonGroup: React.FC<FloatButtonGroupProps> = (props) => {
     [`${groupPrefixCls}-${shape}-shadow`]: !trigger,
   });
 
+  // ============================ zIndex ============================
+  const [zIndex] = useZIndex('FloatButton', style?.zIndex as number);
+
+  const mergedStyle: React.CSSProperties = { zIndex, ...style };
```
Comment (zombieJ) [inline components/float-button/FloatButtonGroup.tsx:54]: 同上

File: components/float-button/FloatButton.tsx (hunk 1 above)
Comment (li-jia-nan) [reply components/float-button/FloatButton.tsx]: done

File: components/float-button/FloatButtonGroup.tsx (hunk 2 above)
Comment (li-jia-nan) [reply components/float-button/FloatButtonGroup.tsx]: done

File: components/float-button/FloatButtonGroup.tsx [hunk 3]
```diff
@@ -47,6 +48,11 @@ const FloatButtonGroup: React.FC<FloatButtonGroupProps> = (props) => {
     [`${groupPrefixCls}-${shape}-shadow`]: !trigger,
   });
 
+  // ============================ zIndex ============================
+  const [zIndex] = useZIndex('FloatButton', style?.zIndex as number);
+
+  const mergedStyle: React.CSSProperties = { ...style, zIndex };
```
Comment (afc163) [inline components/float-button/FloatButtonGroup.tsx:54]: style 权重应该更高吧

Comment (li-jia-nan) [reply components/float-button/FloatButtonGroup.tsx:54]: image

Comment (li-jia-nan) [reply components/float-button/FloatButtonGroup.tsx:54]: 我本来是这样写的

Comment (afc163) [reply components/float-button/FloatButtonGroup.tsx:54]: 哦上面已经给过值了，那应该没问题。

**Review by afc163 — APPROVED** [review]

**Pull request merged**